compiler-rt/lib/scudo/standalone/secondary.h

```diff
@@ -659,8 +711,18 @@ MapAllocator<Config>::tryAllocateFromCache(const Options &Options, uptr Size,
                                            FillContentsMode FillContents) {
   CachedBlock Entry;
   uptr EntryHeaderPos;
+  uptr MaxAllowedFragmentedBytes;
+  const uptr PageSize = getPageSizeCached();
+
+  if (UNLIKELY(useMemoryTagging<Config>(Options))) {
```

Comment:
I don't think you need the UNLIKELY since this is a configuration parameter so it will be evaluated at compile time. The UNLIKELY can be confusing since it's applied to something that is alway true/false.

Reply:
Agree. Sorry I just copied/pasted in the previous comment

Reply:
Correct me if I'm wrong, but can't MTE be turned on and off dynamically?

Reply:
Update: given that the UNLIKELY annotation wasn't used originally in the retrieval code, I'll remove it for now anyway.

Reply:
Actually you are correct, this is checking if memory tagging is enabled.

I would actually flip this and make the negative case first. I know I said that usually the positive case comes first, but I also think the more likely case should come first.

Reply:
Ok, should I reintroduce the LIKELY annotation or just flip the conditional order?

Reply:
I don't think the LIKELY is that necessary, but it does indicate what we think is the normal case. So if you want, feel free.

Reply:
I gave the suggestion to make UNLIKELY come first. In fact, I'm fine with either way. Please follow @cferris1000's suggestion

Reply:
Ok, I will just leave it in, but I agree that it doesn't make much of a difference.

Reply:
Sorry to give conflicting advice here. I would really like to move a lot of this mte code to make it easier to understand what's the normal path, but that's a project for the future.